### PR TITLE
Add support for namespaced DC drivers

### PR DESCRIPTION
**Pull request by Toflar**
Requiring custom DC's to reside in the global namespace is still kind of PITA as you'll have to register that class differently from everything else in your bundle in your `composer.json` autoload section.

Of course, we'll have to come up with an alternative via DI container eventually. But let's at least support placing it in a proper namespace like everything else until then.

Usage:

```php
use Vendor\Driver\FoobarDriver;

$GLOBALS['TL_DCA']['tl_foobar']['config']['dataContainer'] = FoobarDriver::class;
```

### CODE REVIEW

File: core-bundle/src/Resources/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -1388,6 +1388,24 @@ abstract public function getPalette();
 	 * @throws \Exception
 	 */
 	abstract protected function save($varValue);
+
+	/**
+	 * Return the class name for .
```
Comment (fritzmg) [inline core-bundle/src/Resources/contao/classes/DataContainer.php]:
```suggestion
	 * Return the class name of the DataContainer driver for the given table.
```

**Comment by aschempp** [comment]
How about using a factory (`DataContainer::create`) instead?

**Review by ausi — DISMISSED** [review]

**Comment by Toflar** [comment]
Can't use a factory. The constructor params are not standardized across the data containers.

**Comment by leofeyer** [comment]
This is very nice. If you fix the CI chain, I'll happily merge the PR. 👍🏻

**Review by bytehead — DISMISSED** [review]

**Comment by Toflar** [comment]
Done.

**Review by leofeyer — APPROVED** [review]

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Comment by ausi** [comment]
Did we forget to backport this for Contao 4.9?

**Comment by fritzmg** [comment]
I don't think it was "forgotten" but merely regarded as a new feature. But @ausi and me were discussing whether we should backport this simple change to the current LTS, since it truly is a PITA to actually get a driver to be registered otherwise, if you want to follow modern programming principles. @contao/developers wdyt?

**Comment by leofeyer** [comment]
Ok with me.

**Pull request merged**